packages/core/saved-objects/core-saved-objects-server-internal/src/routes/import.ts

```diff
@@ -36,6 +36,8 @@ export const registerImportRoute = (
     {
       path: '/_import',
```

Comment:
I just realised, we probably do not want to mark all to the SO APIs as public in this way as this would make the "public" for serverless too (something we'd like to avoid since they are deprecated and soon to be deleted).

We should only do this for the `_import` and `_export` APIs.

Reply:
> discontinued before we migrate to these docs this summer

These are planned to be removed in 9.0.

Due to the rather sensitive nature of saved object CRUD APIs (they directly expose our database schemas) and how differently database migrations work on serverless (they're fully managed) we should avoid including the SO CRUD concepts in Serverless server documentation and not make them `public` access on serverless.

What I think we could do is add check for `isServerless` and only set them to `public` if not serverless. That way we would be in line with the existing stack docs for onprem but diverge for serverless. This would be best done as a separate PR IMO as we'd also want to add some tests. I'd be curious to here @lukeelmers 's opinion here too.

`_import` and `_export` don't share these same issues.

Reply:
I just created a serverless project and tried to use SO APIs and got the following:

```sh
❯ curl -sH 'authorization: ApiKey cFZHNHVZOEJhMkM0STBQaldfY286aUpIRElvSWxRdENFYUVzb19MZ0s0Zw==' https://test-project-jl-f6b8c6.kb.us-east-1.aws.elastic.cloud/api/saved_objects/dashboard/eb48ada0-df24-4265-8292-f599cdbba741 | jq
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "uri [/api/saved_objects/dashboard/eb48ada0-df24-4265-8292-f599cdbba741] with method [get] exists but is not available with the current configuration"
}
```

Reply:
Thanks for the explanation!  I'm happy to code these (and the rest of the stateful-only APIs) however necessary to give an accurate list of APIs for each context. I'll put this PR back in draft state until we have a way to mark things conditionally public.

Reply:
I agree we should remove documentation for these from serverless - even if they don't end up being removed in 9.0, they will remain internal on serverless, which means users should not be interacting with them.

Reply:
I've added b126d717d9d74839b32f998abfb1618399c643ed to split off a serverless-specific manually-maintained openAPI document with only the import and export endpoints in it.  This is a stop-gap to replace what we're currently using for the serverless API docs.